### ISSUE TITLE
Commit hangs interface with commit button remaining disabled indefinitely

### ISSUE THREAD

**Issue by Mark-Hetherington** [desktop/desktop#11597]
### Describe the bug

When commiting changes, the commit button title changes to "comitting to <branch-name>" with the progress spinner. This seems to remain in place indefinitely (at least 10 minutes for a single line commit) rather than promptly returning to accept more changes. Closing github desktop and reopening it shows the commit has occurred immediately, and other parts of the interface ("Push origin") will update to reflect the commit if the window loses and regains focus.

### Version & OS

Github Desktop Version 2.6.3, Windows 10 home 20H2

### Steps to reproduce the behavior

1. Make a change in the repository, and select changed file
2. Enter a commit summary, and click "commit to branch"
3. Commit process visually appears to keep running indefinitely
4. Closing and reopening Github Desktop reveals an intact commit which was promptly committed.

### Expected behavior

The commit button should promptly revert to it's enabled state, without the progress spinner.

### Actual behavior

The commit to branch button remains disabled, indefinitely, with the loading/progress spinner displayed.

There appears to be little relevant information in the logs, although the timing reflects long processing time:
2021-02-15T23:10:52.867Z - info: [ui] Executing createCommit: git commit -F - (took 285.587s)
2021-02-15T23:10:53.786Z - info: [ui] [Timing] Action 'create commit' for 'kms-sites' took 286.694s

There is also an error on startup:
2021-02-15T22:05:06.093Z - error: [main] Malformed launch arguments received: C:\Users\mark\AppData\Local\GitHubDesktop\app-2.6.3\GitHubDesktop.exe,--allow-file-access-from-files

**Comment by sergiou87** [comment]
Hi @Mark-Hetherington! 👋 

I'm sorry you're experiencing this issue 😱  I don't think that error on startup should worry us. Regarding the commit issue…

> 2021-02-15T23:10:52.867Z - info: [ui] Executing createCommit: git commit -F - (took 285.587s)

This line reflects how long the actual `git commit` command took: https://github.com/desktop/desktop/blob/79c10bf0971d7555d346ba47a03c8b9aa44c8ac7/app/src/lib/git/commit.ts#L26-L33

> 2021-02-15T23:10:53.786Z - info: [ui] [Timing] Action 'create commit' for 'kms-sites' took 286.694s

This other line reflects how the whole process took: starting when you click on the Commit button, and finishing when the `git commit` invocation has finished and the UI updated with the new state of the repository: https://github.com/desktop/desktop/blob/8147c748e79a4f0b8f6b1985b7df0144f6cf2aed/app/src/ui/changes/commit-message.tsx#L233-L235

So if you subtract the time it took to run `git commit`, only an additional 1.1 seconds after `git commit` was used to refresh the repository info and the UI. What all this means is that for some reason `git commit` is taking very long to run, and GitHub Desktop can't do anything about it.

Questions:
1. Have you experienced this problem with previous versions of GitHub Desktop?
2. Do you have some kind of antivirus or something that could slow everything down?
3. Have you tried committing files from the command line to see if `git commit` is also that slow?

Thank you for your patience, I hope we can find out the reason for this problem.

**Comment by Mark-Hetherington** [comment]
1. I've started experiencing it in the past few weeks.
2. Only the built in Windows antivirus, and I don't know of any changes there.
3. Git commit on the command line is fine. 

I'm not sure if the logging I pulled for that specific git commit timing reflects the normal issue. As noted originally, if I close github and re-open it, it becomes clear the commit had already completed. That is to say, my general experience of it is that it will sit waiting to commit longer than I am willing to wait. However if I instead commit, then immediately close the github desktop window (while the progress spinner is still spinning) and re-open it, the commit has already applied.

This morning, after a windows update restart, I've attempted to retest and not observed the same behaviour.  So I'm going to assume there was something in the background on the system causing the issue and not Github desktop.
